services/wallet/router.go

```diff
@@ -0,0 +1,120 @@
+package wallet
+
+import (
+	"context"
+	"math"
+	"math/big"
+	"sync"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/status-im/status-go/params"
+	"github.com/status-im/status-go/services/wallet/async"
+	"github.com/status-im/status-go/services/wallet/chain"
+)
+
+type SuggestedRoutes struct {
+	Networks []params.Network `json:"networks"`
+}
+
+func NewRouter(s *Service) *Router {
+	return &Router{s}
+}
+
+type Router struct {
+	s *Service
+}
+
+func (r *Router) suitableTokenExists(ctx context.Context, network *params.Network, tokens []*Token, account common.Address, amount float64, tokenSymbol string) (bool, error) {
+	for _, token := range tokens {
+		if token.Symbol != tokenSymbol {
+			continue
+		}
+
+		clients, err := chain.NewClients(r.s.rpcClient, []uint64{network.ChainID})
+		if err != nil {
+			return false, err
+		}
+
+		balance, err := r.s.tokenManager.getBalance(ctx, clients[0], account, token.Address)
+		if err != nil {
+			return false, err
+		}
+
+		amountForToken, _ := new(big.Float).Mul(big.NewFloat(amount), big.NewFloat(math.Pow10(int(token.Decimals)))).Int(nil)
+		if balance.Cmp(amountForToken) >= 0 {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+func (r *Router) suggestedRoutes(ctx context.Context, account common.Address, amount float64, tokenSymbol string) (*SuggestedRoutes, error) {
+	areTestNetworksEnabled, err := r.s.accountsDB.GetTestNetworksEnabled()
+	if err != nil {
+		return nil, err
+	}
+
+	networks, err := r.s.rpcClient.NetworkManager.Get(false)
+	if err != nil {
+		return nil, err
+	}
+
+	var (
+		group      = async.NewAtomicGroup(ctx)
+		mu         sync.Mutex
+		candidates = make([]params.Network, 0)
+	)
+	for networkIdx := range networks {
+		network := networks[networkIdx]
+		if network.IsTest != areTestNetworksEnabled {
```

Comment:
@alaibe is this condition what you really want to achieve? 
That condition means you're adding network in the following 2 cases:
- if it is a test network and test networks are enabled 
- if it is not a test network and test networks are not enabled 

but I think you should add it in the following case as well:
- if it is not a test network and test networks are enabled

if above is true then condition should be like:
```
if network.IsTest && !areTestNetworksEnabled {
    continue
}
```

Reply:
> if it is not a test network and test networks are enabled

What do think I should add this:
I don t want to make transaction on mainnet if test networks are enable.
If test networks are enable, I only want to transfer on ropsten/rinkeby...